mailpoet/lib/API/MP/v1/Segments.php

```diff
@@ -27,31 +39,125 @@ public function getAll(): array {
   }
 
   public function addList(array $data): array {
-    if (empty($data['name'])) {
+    $this->validateSegmentName($data);
+
+    try {
+      $segment = $this->segmentsRepository->createOrUpdate($data['name'], $data['description'] ?? '');
+    } catch (\Exception $e) {
       throw new APIException(
-        __('List name is required.', 'mailpoet'),
-        APIException::LIST_NAME_REQUIRED
+        __('The list couldn’t be created in the database', 'mailpoet'),
+        APIException::FAILED_TO_SAVE_LIST
       );
     }
 
-    if (!$this->segmentsRepository->isNameUnique($data['name'], null)) {
+    return $this->buildItem($segment);
+  }
+
+  public function updateList(array $data): array {
+    // firstly validation on list id
+    $this->validateSegmentId((string)($data['id'] ?? ''));
+
+    // secondly validation on list name
+    $this->validateSegmentName($data);
+
+    try {
+      $segment = $this->segmentsRepository->createOrUpdate(
```

Comment:
I was able to update segments that you can not update on the interface, like 'WooCommerce customers.
Is this expected?

```
wp> \MailPoet\API\API::MP('v1')->updateList(['id'=>2 ,'name'=>'superWoo'])
=> array(7) {
  ["id"]=>
  string(1) "2"
  ["name"]=>
  string(8) "superWoo"
  ["type"]=>
  string(17) "woocommerce_users"
  ["description"]=>
  string(0) ""
  ["created_at"]=>
  string(19) "2021-07-29 12:34:17"
  ["updated_at"]=>
  string(19) "2022-10-28 00:28:01"
  ["deleted_at"]=>
  NULL
}
```

`updateList` allows you to update dynamic segments, `deleteList` does not let you delete dynamic segments.
Not sure which one is correct. If this is expected, let's add a note on the documentation.

Reply:
I will add checking the list type. 👍